staging/src/kubevirt.io/client-go/apis/flavor/v1alpha1/types.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// VirtualMachineFlavor - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+type VirtualMachineFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineFlavorSpec `json:"spec" valid:"required"`
+}
+
+// VirtualMachineFlavorList - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type VirtualMachineFlavorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualMachineFlavor `json:"items"`
+}
+
+// VirtualMachineClusterFlavor - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+type VirtualMachineClusterFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineFlavorSpec `json:"spec" valid:"required"`
+}
+
+// VirtualMachineClusterFlavorList - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type VirtualMachineClusterFlavorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualMachineClusterFlavor `json:"items"`
+}
+
+// +k8s:openapi-gen=true
+type VirtualMachineFlavorSpec struct {
+	DefaultProfile VirtualMachineFlavorProfile         `json:"defaultProfile"`
+	CustomProfiles []VirtualMachineFlavorCustomProfile `json:"customProfiles,omitempty"`
+}
```

Comment:
When I think about how to implement this feature, I envision the merging of the DefaultProfile with the CustomProfile as something that might be tedious to implement/test. One thought that might help get code merged faster in incremental steps would be to just implement the DefaultProfile usage first, then follow in a separate PR the customProfile usage.

do whatever makes the most sense to you, i just thought it would be worth throwing that out as an option.

Reply:
I'm not sure why it would be tedious. I had the implementation already written when you wrote this comment, so I did not change it.
Is it ok this way?

Reply:
yeah, absolutely. it's whatever is easiest for you. I was just pointing this out in case you were looking for ways to reduce the scope of the initial pr and break the implementation into multiple PR